RecoEgamma/EgammaTools/plugins/HGCalElectronIDValueMapProducer.cc

```diff
@@ -0,0 +1,238 @@
+
+      // Energies / PT
+      maps_["gsfTrackPt"].push_back(electron.gsfTrack()->pt());
+      maps_["pOutPt"].push_back(std::sqrt(electron.trackMomentumAtVtx().perp2()));
+      maps_["scEt"].push_back(electron.superCluster()->energy() / std::cosh(electron.superCluster()->eta()));
+      maps_["scEnergy"].push_back(electron.superCluster()->energy());
+      maps_["ecOrigEt"].push_back(electron.electronCluster()->energy() / std::cosh(electron.electronCluster()->eta()));
+      maps_["ecOrigEnergy"].push_back(electron.electronCluster()->energy());
+
+      // energies calculated in an cylinder around the axis of the electron cluster
+      float ec_tot_energy = ld.energyEE() + ld.energyFH() + ld.energyBH();
+      maps_["ecEt"].push_back(ec_tot_energy / std::cosh(electron.electronCluster()->eta()));
+      maps_["ecEnergy"].push_back(ec_tot_energy);
+      maps_["ecEnergyEE"].push_back(ld.energyEE());
+      maps_["ecEnergyFH"].push_back(ld.energyFH());
+      maps_["ecEnergyBH"].push_back(ld.energyBH());
+
+      // Track-based
+      maps_["fbrem"].push_back( electron.fbrem() );
+      maps_["gsfTrackHits"].push_back( electron.gsfTrack()->hitPattern().trackerLayersWithMeasurement() );
+      maps_["gsfTrackChi2"].push_back( electron.gsfTrack()->normalizedChi2() );
+
+      reco::TrackRef myTrackRef = electron.closestCtfTrackRef();
+      bool validKF = myTrackRef.isAvailable() && myTrackRef.isNonnull();
+      maps_["kfTrackHits"].push_back( (validKF) ? myTrackRef->hitPattern().trackerLayersWithMeasurement() : -1 );
+      maps_["kfTrackChi2"].push_back( (validKF) ? myTrackRef->normalizedChi2() : -1 );
+
+      // Track-matching
+      maps_["dEtaTrackClust"].push_back( electron.deltaEtaEleClusterTrackAtCalo() );
+      maps_["dPhiTrackClust"].push_back( electron.deltaPhiEleClusterTrackAtCalo() );
+
+      // Cluster shapes
+      // PCA related
+      maps_["pcaEig1"].push_back(eIDHelper_->eigenValues()(0));
+      maps_["pcaEig2"].push_back(eIDHelper_->eigenValues()(1));
+      maps_["pcaEig3"].push_back(eIDHelper_->eigenValues()(2));
+      maps_["pcaSig1"].push_back(eIDHelper_->sigmas()(0));
+      maps_["pcaSig2"].push_back(eIDHelper_->sigmas()(1));
+      maps_["pcaSig3"].push_back(eIDHelper_->sigmas()(2));
+
+      // transverse shapes
+      maps_["sigmaUU"].push_back(eIDHelper_->sigmaUU());
+      maps_["sigmaVV"].push_back(eIDHelper_->sigmaVV());
+      maps_["sigmaEE"].push_back(eIDHelper_->sigmaEE());
+      maps_["sigmaPP"].push_back(eIDHelper_->sigmaPP());
+
+
+      // long profile
+      maps_["nLayers"].push_back(ld.nLayers());
+      maps_["firstLayer"].push_back(ld.firstLayer());
+      maps_["lastLayer"].push_back(ld.lastLayer());
+      maps_["e4oEtot"].push_back(ld.e4oEtot());
+      maps_["layerEfrac10"].push_back(ld.layerEfrac10());
+      maps_["layerEfrac90"].push_back(ld.layerEfrac90());
+      //maps_["firstLayerEnergy"].push_back(ld.energyPerLayer()[ld.firstLayer()]);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant